nixos/modules/services/home-automation/matter-server.nix

```diff
@@ -75,15 +75,15 @@ in
         TemporaryFileSystem = "/";
         # Allowlist /nix/store (to allow the binary to find its dependencies)
         # and dbus.
-        ReadOnlyPaths = "/nix/store /run/dbus";
+        BindReadOnlyPaths = "/nix/store /run/dbus";
         # Let systemd manage `/var/lib/matter-server` for us inside the
         # ephemeral TemporaryFileSystem.
         StateDirectory = storageDir;
         # `python-matter-server` writes to /data even when a storage-path is
         # specified. This bind-mount points /data at the systemd-managed
         # /var/lib/matter-server, so all files get dropped into the state
         # directory.
-        BindPaths = "${storagePath}:/data";
+        BindPaths = "/var/lib/private/${storageDir}:/data:rbind";
```

Comment:
Does systemd offer a % variable for this? I only know %h from the top of my mind which is the home directory

Reply:
It looks like [%S](https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html) is a short variable for /var/lib; however, I still experienced issues with this approach so we have taken a different approach in the latest iteration.